os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c

```diff
@@ -716,10 +716,15 @@ static int _tizenrt_create_task(struct task_struct *ptask, const char *name, u32
 	return _FAIL;
 #else
 	int func_addr, ctx_addr;
-	pid_t pid;
 	char str_func_addr[9];
 	char str_ctx_addr[9];
 	char *task_info[3];
+
+	pthread_t *pid = (pthread_t *)_tizenrt_zmalloc(sizeof(pthread_t));
```

Comment:
I'm not sure why you changed like this.
It is same as previous, but looks not good.
And if you want to use like this, please add `free` for failure or termination case.

Reply:
You are right Mr Kim, I fix it in new commit. We made these changes to fix warings of incompatible type. Basically I followed the way in #if USE_PTHREAD_MUTEX in the same API itself.

Reply:
That's just integer value. You don't need to allocate the memory for that. This wastes the memory. And I can't find the free at success case. I mean when you delete or finish the task, there is no free for this allocation. (What you added is a failure case.)

Reply:
Thank you Mr Chang. Let me clarify on that: 
1. You are right, if I want to use malloc, I still need to modify _tizenrt_delete_task() to free the memory when deleting the task.
2. Why I made those changes are because in os\board\rtl8721csm\src\component\os\os_dep\include\osdep_service.h, line 182, for the struct task_struct, _thread_hdl_ task is actually a pointer which we assign value like this: ptask->task = pid; which is not correct if pid is an integer.

So do you think I can proceed to fix it by modifying _tizenrt_delete_task()?

Reply:
1. If you use the allocation way, it wastes memory. (Actually it requires an integer memory, but allocation way requires two integer memory.)
2. I don't think you can know the time to be deleted for all of threads you made. Then how will you call the delete function?

Reply:
1. Dear Mr Chang. From my point of view, assigning an integer to a pointer is wrong usage. And that's why we need to fix it. So may I ask your advice? 
2. I understand that if we do not delete task on time, which might cause memory leakage with my modifications. But even with current branch's code, if we do not delete task, calling task_delete(pid), we still have other issues right? So for both case we need to make sure task creation and deletion are properly handled.

Reply:
1. PID is an integer value always. You don't need to keep it as pointer value. In my opinion, you should change the type in your structure.
```
typedef pid_t _thread_hdl_;
```
or
```
struct task_struct {
	const char *task_name;
	pid_t task;	/* I: workqueue thread */
```
2. Because you use a pointer and allocation way for pid value, task termination has a problem. If you use my suggestion as 1st, then there is no issue.
```
But even with current branch's code, if we do not delete task, calling task_delete(pid), we still have other issues right? So for both case we need to make sure task creation and deletion are properly handled.
```
In your code, yes. But you can't modify it with current pointer type I think.

Reply:
Sure Mr Chang, I checked our code, and changed our struct as adviced.